src/hdf5_hl.ts

```diff
@@ -421,26 +482,19 @@ abstract class HasAttrs {
   get attrs() {
     let attr_names = Module.get_attribute_names(this.file_id, this.path) as string[];
     let attrs: {[key: string]: Attribute}  = {};
+    const { file_id, path } = this;
     for (let name of attr_names) {
-      let metadata = Module.get_attribute_metadata(this.file_id, this.path, name);
       Object.defineProperty(attrs, name, {
-        get: (): Attribute => ({
-          value: get_attr(this.file_id, this.path, name),
-          shape: metadata.shape,
-          dtype: metadata_to_dtype(metadata),
-          metadata
-        }),
+        get: (): Attribute => (new Attribute(file_id, path, name)),
         enumerable: true
       });
-
-      //attrs[name] = get_attr(this.file_id, this.path, name);
     }
     return attrs;
 
   }
 
-  get_attribute(name: string) {
-    get_attr(this.file_id, this.path, name);
+  get_attribute(name: string, json_compatible: boolean) {
```

Comment:
To keep the behaviour consistent, I would propose to have `json_compatible` optional and set to `false` by default (as you did for `process_data`)